### PR TITLE
[0285] ShowConstantTBT Description Update

### LINKED ISSUE #1573: [SDL 0285] ShowConstantTBT Description Update
Proposal: [ShowConstantTBT Description Update](https://github.com/smartdevicelink/sdl_evolution/edit/master/proposals/0285-ConstantTBT-Update.md)

> This proposal is to update the descriptions of `ShowConstantTBT`'s parameters `distanceToManeuver` and `distanceToManeuverScale` so we get consistent use of them across all OEMs and Navigation apps.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/955

The following comments need to be updated: 
- `ShowConstantTBT.distanceToManeuver`
- `ShowConstantTBT.distanceToManeuverScale`

### PR DESCRIPTION
**Pull request by kostyaBoss**
Fixes #1573 

This PR is **[ready]** for review.

### Related PR's
[Core](https://github.com/smartdevicelink/sdl_core/pull/3596)

### Risk
This PR makes **[minor]** API changes.

### Testing Plan
- [x] I have run the unit tests with this PR
- [x] I have tested this PR against Core and verified the behavior
- [x] I have tested Android

#### Unit Tests
Added unit tests for the new structures

### Summary
Changes according to #1573 

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by kostyaBoss** [comment]
@santanamk ready for the Ford review

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/ShowConstantTbtTests.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import com.smartdevicelink.protocol.enums.FunctionID;
```
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/ShowConstantTbtTests.java:4]: @kostyaBoss I would revert these changes in `ShowConstantTbtTests.java`.  They are valid changes, but are out of scope of the PR.

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/responses/ShowConstantTbtResponseTests.java [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import com.smartdevicelink.marshal.JsonRPCMarshaller;
 import com.smartdevicelink.protocol.enums.FunctionID;
 import com.smartdevicelink.proxy.RPCMessage;
```
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/responses/ShowConstantTbtResponseTests.java:5]: @kostyaBoss I would revert these changes in `ShowConstantTbtResponseTests.java`.  This PR only deals with the `ShowConstantTbtRequest` so these changes are out of scope of the PR.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbtResponse.java [hunk 3]
```diff
@@ -44,23 +44,31 @@
  *
  * @since SmartDeviceLink 2.0
  */
-public class ShowConstantTbtResponse extends RPCResponse {
```
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbtResponse.java:47]: @kostyaBoss I would revert these changes in `ShowConstantTbtResponse.java`.  This PR only deals with the `ShowConstantTbtRequest` so these changes are out of scope of the PR.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java [hunk 4]
```diff
@@ -49,34 +50,33 @@
...
  * @since SmartDeviceLink 2.0
  */
-public class ShowConstantTbt extends RPCRequest {
+public class ShowConstantTBT extends RPCRequest {
 
     public static final String KEY_TEXT1 = "navigationText1";
     public static final String KEY_TEXT2 = "navigationText2";
     public static final String KEY_ETA = "eta";
     public static final String KEY_TOTAL_DISTANCE = "totalDistance";
-    public static final String KEY_MANEUVER_DISTANCE = "distanceToManeuver";
```
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java:58]: @kostyaBoss Lines 59-60 and 206-252 are good.

I would revert the other changes, since they are out of scope of the PR.

**Review by santhanamk — COMMENTED** [review]
@kostyaBoss I have reviewed this PR.  The code looks good.  Please review whenever you get the chance, and make changes as needed.

**Comment by kostyaBoss** [comment]
@santhanamk implementing experience gained while fixing the 0236 - this will cause the crash of the generic tests, which are testing the structures under the hood. Do we still need to rollback these changes? (they are according to XML provided)

**Comment by santhanamk** [comment]
> @santhanamk implementing experience gained while fixing the 0236 - this will cause the crash of the generic tests, which are testing the structures under the hood. Do we still need to rollback these changes? (they are according to XML provided)

@kostyaBoss I would roll back the changes in: 
`ShowConstantTbtTests.java`,
`ShowConstantTbtResponseTests.java`,
`ShowConstantTbtResponse.java`.

I believe that the changes in these three files are out of the scope of the PR.

I went through and reviewed what needs to be changed in `ShowConstantTbt.java`.  I would not rename the file since it is out of scope of the PR.

**Review by santhanamk — COMMENTED** [review]
@kostyaBoss I have reviewed the PR.  Please take a look at the suggestions, and make the changes as needed whenever you get the chance.

**Comment by kostyaBoss** [comment]
@santhanamk I am sorry for the big delay providing this answer but if we will rollback these changes, what we know from the other pr's - the generic internal tests, which are testing the RPC in the documentation will fail. These changes - according to XML file. Do we still need to rollback?

**Comment by santhanamk** [comment]
> @santhanamk I am sorry for the big delay providing this answer but if we will rollback these changes, what we know from the other pr's - the generic internal tests, which are testing the RPC in the documentation will fail. These changes - according to XML file. Do we still need to rollback?

@kostyaBoss So I spoke with Michael Crimando about this.  He wrote the proposal.

This is purely a description change, and nothing else.  You just need to change the descriptions for those two parameters `distanceToManeuver` and `distanceToManeuverScale` in `ShowConstantTbt.java`.  No other changes are needed.

**Comment by kostyaBoss** [comment]
@santhanamk I've updated the pr, please, re-review

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/ShowConstantTbtTests.java [hunk 5]
```diff
@@ -30,7 +30,7 @@
 
 /**
  * This is a unit test class for the SmartDeviceLink library project class :
- * {@link com.smartdevicelink.proxy.rpc.ShowConstantTbt}
+ * {@link ShowConstantTbt}
```
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/ShowConstantTbtTests.java:33]: @kostyaBoss On line 33 I would remove it and change it back to `{@link com.smartdevicelink.proxy.rpc.ShowConstantTbt}`

The idea is that nothing should be changed in this class.

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/ShowConstantTbtTests.java [hunk 6]
```diff
@@ -71,8 +71,8 @@ protected JSONObject getExpectedParameters(int sdlVersion) {
             result.put(ShowConstantTbt.KEY_SOFT_BUTTONS, TestValues.JSON_SOFTBUTTONS);
             result.put(ShowConstantTbt.KEY_ETA, TestValues.GENERAL_STRING);
             result.put(ShowConstantTbt.KEY_MANEUVER_COMPLETE, true);
-            result.put(ShowConstantTbt.KEY_MANEUVER_DISTANCE, TestValues.GENERAL_DOUBLE);
-            result.put(ShowConstantTbt.KEY_MANEUVER_DISTANCE_SCALE, TestValues.GENERAL_DOUBLE);
+            result.put(ShowConstantTbt.KEY_DISTANCE_TO_MANEUVER, TestValues.GENERAL_DOUBLE);
```
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/ShowConstantTbtTests.java:74]: @kostyaBoss On line 74 I would rename `KEY_DISTANCE_TO_MANEUVER` back to its old variable name `KEY_MANEUVER_DISTANCE`.  You can apply this change throughout the class.

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/ShowConstantTbtTests.java [hunk 7]
```diff
@@ -71,8 +71,8 @@ protected JSONObject getExpectedParameters(int sdlVersion) {
...
+            result.put(ShowConstantTbt.KEY_DISTANCE_TO_MANEUVER_SCALE, TestValues.GENERAL_DOUBLE);
```
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/ShowConstantTbtTests.java:75]: @kostyaBoss On line 75 I would rename `KEY_DISTANCE_TO_MANEUVER_SCALE` back to its old variable name `KEY_MANEUVER_DISTANCE_SCALE`.  You can apply this change throughout the class.

File: android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/responses/ShowConstantTbtResponseTests.java [hunk 8]
```diff
@@ -21,7 +21,7 @@
 
 /**
  * This is a unit test class for the SmartDeviceLink library project class :
- * {@link com.smartdevicelink.proxy.rpc.ShowConstantTbtResponse}
+ * {@link ShowConstantTbtResponse}
```
Comment (santhanamk) [inline android/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/responses/ShowConstantTbtResponseTests.java:24]: @kostyaBoss I would delete line 24 and replace it with its old line `{@link com.smartdevicelink.proxy.rpc.ShowConstantTbtResponse}`

File: base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java [hunk 9]
```diff
@@ -33,6 +33,7 @@
 
 import com.smartdevicelink.protocol.enums.FunctionID;
 import com.smartdevicelink.proxy.RPCRequest;
+import com.smartdevicelink.util.SdlDataTypeConverter;
```
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java:36]: @kostyaBoss I would remove this import on line 36 `import com.smartdevicelink.util.SdlDataTypeConverter;`.

(The idea is to just make the description changes and nothing else)

File: base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java [hunk 10]
```diff
@@ -55,8 +56,8 @@
     public static final String KEY_TEXT2 = "navigationText2";
     public static final String KEY_ETA = "eta";
     public static final String KEY_TOTAL_DISTANCE = "totalDistance";
-    public static final String KEY_MANEUVER_DISTANCE = "distanceToManeuver";
-    public static final String KEY_MANEUVER_DISTANCE_SCALE = "distanceToManeuverScale";
+    public static final String KEY_DISTANCE_TO_MANEUVER = "distanceToManeuver";
```
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java:59]: @kostyaBoss On line 59 I would rename this variable back to its old name `KEY_MANEUVER_DISTANCE`.  You can apply this change throughout the class.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java [hunk 11]
```diff
@@ -55,8 +56,8 @@
...
+    public static final String KEY_DISTANCE_TO_MANEUVER_SCALE = "distanceToManeuverScale";
```
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java:60]: @kostyaBoss On line 60 I would rename this variable back to its old name `KEY_MANEUVER_DISTANCE_SCALE`.  You can apply this change throughout the class.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java [hunk 12]
```diff
@@ -203,45 +204,53 @@ public Image getNextTurnIcon() {
...
      *
-     * @return Double -a Double value representing a Distance till next maneuver (starting from) from previous maneuver
+     * @return Float Distance (in meters) from previous maneuver to next maneuver. May be used to calculate
+     * progress bar.
+     * {"num_min_value": 0.0, "num_max_value": 1000000000.0}
+     * @since SmartDeviceLink 2.0.0
      */
     public Double getDistanceToManeuverScale() {
-        return getDouble(KEY_MANEUVER_DISTANCE_SCALE);
+        Object object = getParameters(KEY_DISTANCE_TO_MANEUVER_SCALE);
```
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java:252]: @kostyaBoss  I would delete lines 252-253 and replace it with the old code `return getDouble(KEY_MANEUVER_DISTANCE_SCALE);`.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbtResponse.java [hunk 13]
```diff
@@ -46,19 +46,27 @@
  */
 public class ShowConstantTbtResponse extends RPCResponse {
 
+    /**
```
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbtResponse.java:49]: @kostyaBoss I would delete the comment block on lines 49-51.  The idea is that nothing should be changed in this file.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbtResponse.java [hunk 14]
```diff
@@ -46,19 +46,27 @@
...
+     * Constructs a new ShowConstantTBTResponse object
+     */
     public ShowConstantTbtResponse() {
         super(FunctionID.SHOW_CONSTANT_TBT.toString());
     }
 
+    /**
```
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbtResponse.java:56]: @kostyaBoss I would delete the comment block on lines 56-60.  The idea is that nothing should be changed in this file.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbtResponse.java [hunk 15]
```diff
@@ -46,19 +46,27 @@
...
     public ShowConstantTbtResponse(Hashtable<String, Object> hash) {
         super(hash);
     }
 
     /**
      * Constructs a new ShowConstantTbtResponse object
      *
-     * @param success    whether the request is successfully processed
-     * @param resultCode whether the request is successfully processed
+     * @param success whether the request is successfully processed
```
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbtResponse.java:68]: @kostyaBoss I would delete lines 68-69 and bring back the old formatted code.

File: base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java [hunk 16]
```diff
@@ -203,45 +204,53 @@ public Image getNextTurnIcon() {
...
+     * Gets the distanceToManeuver.
      *
-     * @return Double -a Double value representing a Fraction of distance till next maneuver
+     * @return Float Distance (in meters) until next maneuver. May be used to calculate progress bar.
+     * {"num_min_value": 0.0, "num_max_value": 1000000000.0}
+     * @since SmartDeviceLink 2.0.0
      */
     public Double getDistanceToManeuver() {
-        return getDouble(KEY_MANEUVER_DISTANCE);
+        Object object = getParameters(KEY_DISTANCE_TO_MANEUVER);
```
Comment (santhanamk) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/ShowConstantTbt.java:226]: @kostyaBoss I would delete lines 226 and 227 and replace it with the old code `return getDouble(KEY_MANEUVER_DISTANCE);`.

**Review by santhanamk — COMMENTED** [review]
@kostyaBoss I reviewed the changes for the PR.  I have a few more suggestions.  They are pretty minor.  The idea is just to change the descriptions for the parameters, and nothing else.  Please take a look at the suggestions whenever you get the chance.

**Comment by kostyaBoss** [comment]
@santhanamk rollbacked the required changes according to your request and double-checked everything, all the comments seem to be processed. Please, let me know if some additional changes are required there

**Review by santhanamk — APPROVED** [review]
@kostyaBoss Code looks good. I have approved the PR.

Once you have tested against Core and HMI (if applicable), please put the links for those PRs. Please mark the PR as ready for review, and you can tag Livio once the testing is done on your end.

**Comment by kostyaBoss** [comment]
@bilal-alsharifi , @jordynmackool This pr is ready for the Livio review

**Review by bilal-alsharifi — APPROVED** [review]
